src/CommunityToolkit.Maui/Behaviors/PlatformBehaviors/Touch/GestureManager.cs

```diff
@@ -0,0 +1,828 @@
+		{
+			sender.InteractionStatus = interactionStatus;
+			sender.RaiseInteractionStatusChanged();
+		}
+	}
+
+	internal async ValueTask HandleTouch(TouchBehavior sender, TouchStatus status, CancellationToken token)
+	{
+		if (sender.IsDisabled)
+		{
+			return;
+		}
+
+		var canExecuteAction = sender.CanExecute;
+		if (status != TouchStatus.Started || canExecuteAction)
+		{
+			if (!canExecuteAction)
+			{
+				status = TouchStatus.Canceled;
+			}
+
+			var state = status is TouchStatus.Started
+				? TouchState.Pressed
+				: TouchState.Normal;
+
+			if (status is TouchStatus.Started)
+			{
+				animationProgress = 0;
+				animationState = state;
+			}
+
+			var isToggled = sender.IsToggled;
+			if (isToggled.HasValue)
+			{
+				if (status != TouchStatus.Started)
+				{
+					durationMultiplier = (animationState is TouchState.Pressed && !isToggled.Value) ||
+						(animationState is TouchState.Normal && isToggled.Value)
+							? 1 - animationProgress
+							: animationProgress;
+
+					await UpdateStatusAndState(sender, status, state, token);
+
+					if (status is TouchStatus.Canceled)
+					{
+						await sender.ForceUpdateState(token, false);
+						return;
+					}
+
+					OnTapped(sender);
+					sender.IsToggled = !isToggled;
+					return;
+				}
+
+				state = isToggled.Value
+					? TouchState.Normal
+					: TouchState.Pressed;
+			}
+
+			await UpdateStatusAndState(sender, status, state, token);
+		}
+
+		if (status is TouchStatus.Completed)
+		{
+			OnTapped(sender);
+		}
+	}
+
+	internal async ValueTask HandleHover(TouchBehavior sender, HoverStatus status, CancellationToken token)
+	{
+		if (!sender.Element?.IsEnabled ?? true)
+		{
+			return;
+		}
+
+		var hoverState = status is HoverStatus.Entered
+			? HoverState.Hovered
+			: HoverState.Normal;
+
+		if (sender.HoverState != hoverState)
+		{
+			sender.HoverState = hoverState;
+			await sender.RaiseHoverStateChanged(token);
+		}
+
+		if (sender.HoverStatus != status)
+		{
+			sender.HoverStatus = status;
+			sender.RaiseHoverStatusChanged();
+		}
+	}
+
+	internal async Task ChangeStateAsync(TouchBehavior sender, bool animated, CancellationToken token)
+	{
+		var status = sender.Status;
+		var state = sender.State;
+		var hoverState = sender.HoverState;
+
+		AbortAnimations(sender);
+		animationTokenSource = new CancellationTokenSource();
+
+		var isToggled = sender.IsToggled;
+
+		if (sender.Element is not null)
+		{
+			UpdateVisualState(sender.Element, state, hoverState);
+		}
+
+		if (!animated)
+		{
+			if (isToggled.HasValue)
+			{
+				state = isToggled.Value
+					? TouchState.Pressed
+					: TouchState.Normal;
+			}
+
+			var durationMultiplier = this.durationMultiplier;
+			this.durationMultiplier = null;
+
+			try
+			{
+				await RunAnimationTask(sender, state, hoverState, animationTokenSource.Token, durationMultiplier.GetValueOrDefault()).ConfigureAwait(false);
+			}
+			catch (TaskCanceledException ex)
+			{
+				Trace.WriteLine(ex);				
+			}
```

Comment:
So this one will need some investigation... The `RunAnimationTask` [method](https://github.com/Axemasta/CommunityToolkit.Maui/blob/rebase-touch-effect/src/CommunityToolkit.Maui/Behaviors/PlatformBehaviors/Touch/GestureManager.cs#L747) returns a `Task<bool>`, but we never look the bool value for it, so I would say that we can just return a `Task`. With that's possible, then we can change this code to be

```suggestion
			try
			{
				await RunAnimationTask(sender, state, hoverState, animationTokenSource.Token, durationMultiplier.GetValueOrDefault()).ConfigureAwait(ConfigureAwaitOptions.SuppressThrowing);
			}
			catch (TaskCanceledException ex)
			{
				Trace.WriteLine(ex);				
			}
```